test/functional/wallet_importdescriptors.py

```diff
@@ -0,0 +1,293 @@
+
+        # # Test import fails if no descriptor present
+        key = get_key(w0)
+        self.log.info("Import should fail if a descriptor is not provided")
+        self.test_importdesc({"timestamp": "now"},
+                             success=False,
+                             error_code=-8,
+                             error_message='Descriptor not found.')
+
+        # # Test importing of a P2PKH descriptor
+        key = get_key(w0)
+        self.log.info("Should import a p2pkh descriptor")
+        self.test_importdesc({"desc": descsum_create("pkh(" + key.pubkey + ")"),
+                              "timestamp": "now",
+                              "label": "Descriptor import test"},
+                             success=True)
+        test_address(w1,
+                     key.p2pkh_addr,
+                     solvable=True,
+                     ismine=True,
+                     labels=["Descriptor import test"])
+
+        # # Test importing of a P2SH-P2WPKH descriptor
+        key = get_key(w0)
+        self.log.info("Should not import a p2sh-p2wpkh descriptor without checksum")
+        self.test_importdesc({"desc": "sh(wpkh(" + key.pubkey + "))",
+                              "timestamp": "now"
+                              },
+                             success=False,
+                             error_code=-5,
+                             error_message="Missing checksum")
+
+        self.log.info("Should not import a p2sh-p2wpkh descriptor that has range specified")
+        self.test_importdesc({"desc": descsum_create("sh(wpkh(" + key.pubkey + "))"),
+                               "timestamp": "now",
+                               "range": 1,
+                              },
+                              success=False,
+                              error_code=-8,
+                              error_message="Range should not be specified for an un-ranged descriptor")
+
+        self.log.info("Should not import a p2sh-p2wpkh descriptor and have it set to active")
+        self.test_importdesc({"desc": descsum_create("sh(wpkh(" + key.pubkey + "))"),
+                               "timestamp": "now",
+                               "active": True,
+                              },
+                              success=False,
+                              error_code=-8,
+                              error_message="Active descriptor must be ranged")
+
+        self.log.info("Should import a (non-active) p2sh-p2wpkh descriptor")
+        self.test_importdesc({"desc": descsum_create("sh(wpkh(" + key.pubkey + "))"),
+                               "timestamp": "now",
+                               "active": False,
+                              },
+                              success=True)
+
+        test_address(w1,
+                     key.p2sh_p2wpkh_addr,
+                     ismine=True,
+                     solvable=True)
+
+        # # Test importing of a multisig descriptor
```

Comment:
`add importdescriptors RPC and tests for native descriptor wallets`

We should also test `sh(wsh())` and make sure that the various imports aren't "displacing" the other types. (I didn't see it might have missed it)

Reply:
Not sure what you mean by "displacing"

Reply:
In other words, make sure that importing legacy doesn't accidentally replace p2sh somehow, and p2sh doesn't accidentally also replace bech32, and so on. Just thinking about what could have gone wrong under the hood.

Reply:
Added a test for that case.